cli/util.go

```diff
@@ -0,0 +1,177 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/coderd/autobuild/schedule"
+	"github.com/coder/coder/coderd/util/tz"
+)
+
+var errInvalidScheduleFormat = xerrors.New("Schedule must be in the format Mon-Fri 09:00AM America/Chicago")
+var errInvalidTimeFormat = xerrors.New("Start time must be in the format hh:mm[am|pm] or HH:MM")
+var errUnsupportedTimezone = xerrors.New("The location you provided looks like a timezone. Check https://ipinfo.io for your location.")
+
+// durationDisplay formats a duration for easier display:
+//   * Durations of 24 hours or greater are displays as Xd
+//   * Durations less than 1 minute are displayed as <1m
+//   * Duration is truncated to the nearest minute
+//   * Empty minutes and seconds are truncated
+//   * The returned string is the absolute value. Use sign()
+//     if you need to indicate if the duration is positive or
+//     negative.
+func durationDisplay(d time.Duration) string {
+	duration := d
+	sign := ""
+	if duration == 0 {
+		return "0s"
+	}
+	if duration < 0 {
+		duration *= -1
+	}
+	// duration > 0 now
+	if duration < time.Minute {
+		return sign + "<1m"
+	}
+	if duration > 24*time.Hour {
+		duration = duration.Truncate(time.Hour)
+	}
+	if duration > time.Minute {
+		duration = duration.Truncate(time.Minute)
+	}
+	days := 0
+	for duration.Hours() >= 24 {
+		days++
+		duration -= 24 * time.Hour
+	}
+	durationDisplay := duration.String()
+	if days > 0 {
+		durationDisplay = fmt.Sprintf("%dd%s", days, durationDisplay)
+	}
+	for _, suffix := range []string{"m0s", "h0m", "d0s"} {
+		if strings.HasSuffix(durationDisplay, suffix) {
+			durationDisplay = durationDisplay[:len(durationDisplay)-2]
+		}
+	}
+	return sign + durationDisplay
+}
```

Comment:
review: smaller durations are always <1m now, it can also handle negative durations